src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
-            dbg!(&bytes); // this is correct by my accounts
-            let reader = &*bytes;
-            let proposal = UncheckedProposal::from_request(reader, query, headers).unwrap();
+            // extract original PSBT from HTTP request
+            // TODO: `UncheckedProposal` is not really the proposal, but the original PSBT? Is the naming wrong?
+            // TODO: Unneeded and/or undocumented traits in bip-78 library? (`Proposal`, `Headers`)
+            // TODO: We should really save the original PSBT just in case the sender does not sign
+            //  and broadcast the proposed PSBT, or some other error occurs
+            let headers = Headers(req.headers().clone());
+            let query = req.uri().query().map(ToString::to_string);
+            let body_bytes = dbg!(hyper::body::to_bytes(req.into_body()).await?);
+            let proposal = UncheckedProposal::from_request(&*body_bytes, query.as_deref(), headers)
+                .expect("received invalid proposal - TODO: handle this error properly");
             if proposal.is_output_substitution_disabled() {
-                panic!("Output substitution must be enabled");
+                panic!("Output substitution must be enabled"); // TODO: Proper checks for `pjos`
             }
-
             let proposal = proposal
                 .assume_interactive_receive_endpoint() // TODO Check
                 .assume_no_inputs_owned() // TODO Check
                 .assume_no_mixed_input_scripts() // This check is silly and could be ignored
                 .assume_no_inputs_seen_before(); // TODO
 
+            // `psbt` is going to ACTUALLY be the proposal PSBT
             let mut psbt = proposal.psbt().clone();
-            eprintln!("Received transaction: {:#?}", psbt);
-            {
-                for input in &mut psbt.unsigned_tx.input {
-                    // clear signature
-                    input.script_sig = bitcoin::blockdata::script::Script::new();
-                }
-            }
+            eprintln!("Received transaction: {:#?}", psbt); // but not yet
+            psbt.unsigned_tx
+                .input
+                .iter_mut()
+                .for_each(|txin| txin.script_sig = bitcoin::Script::new());
+
+            // find our output
+            // TODO: Would it be possible that the user sends to multiple outputs that we own?
             let (our_output, scheduled_payjoin) = handler
                 .payjoins
                 .find(&mut psbt.unsigned_tx.output)
                 .expect("the transaction doesn't contain our output");
-            let total_channel_amount: bitcoin::Amount = scheduled_payjoin
+
+            // TODO: make this a member fn of `ScheduledPayJoin`
+            // TODO: make sure it is impossible to create a BIP21pj linked to `ScheduledPayJoin`
+            //  with a "total channel amount" larger than requested + fee
+            let total_channel_amount = scheduled_payjoin
```

Comment:
What's the problem with `total_channel_amount > `Bip78Uri.amount()`? The edge case I imagine is BTCPayServer requests an invoice for a strict amount and the excess is either 1. added to on chain single sig or 2. increase a channel amount. It does not seem like requesting more is dire.